### DISCUSSION TITLE
Support for Traditional Chinese language

### DISCUSSION THREAD

**Discussion by bluefoxlee** [OHF-Voice/intents#417] [Supported Sentences]
As title. although there is Simplified Chinese added already, however there is still a slightly difference between two languages.
Wonder will it be better to separate into two in order to get better management and support.

**Comment by eddychanwork** [comment]
Hi, I can help with translating to zh-hk. However, I notice that Simplified Chinese contributor declared their language tag to zh and we request their to change the tag to zh-cn.

**Comment by cmoo2011** [comment]
Hello, I would like to contribute to zh-tw, which is traditional Chinese used in Taiwan.

I totally agree with the comments above. There surely exist difference between zh-cn, zh-hk, and zh-tw. 
These differences would be especially obvious in spoken language (words sometimes pronounced differently/having different meanings) and thus the language "zh" should be separated.

**Comment by synesthesiam** [comment]
For just matching text, do you think the differences are very large?
I agree for voice (speech to text and text to speech), we will need to separate.

**Reply by cmoo2011** [reply]
Well... they are similar, but different IMHO.

Traditional Chinese and Simplified Chinese(i.e. zh-cn) uses lots of different characters and they are not one-to-one matching. 

In some cases, a character in Simplified Chinese (zh-cn) can be translated in to several different Traditional Chinese characters depending on the context, which also cause some problem to translation softwares.(I've seen some softwares that uses Google translate to convert their zh-cn UI into traditional Chinese... their results are often awkward and unnatural for traditional Chinese users.)


In a nutshell, though Traditional and Simplified Chinese are both “Chinese”, they are different languages for native speakers after all.

**Reply by eddychanwork** [reply]
I think the difference are very large due to our cultural differences. Although simplified Chinese is modified from traditional Chinese, the character might be unrecognisable by traditional Chinese user. Beside, China, Taiwan and Hong Kong usually adopt translation approach.

For example:
English: Camera 
zh-CN: 摄像机
zh-HK: 攝影機

**Comment by bluefoxlee** [comment]
@synesthesiam
I would like to apply and volunteer myself to Chinese language leader role either for T.Chinese (zh_tw) or Chinese (zh, if dev team decide to not to separate) to push the Chinese part moving.I am currently contributing to Home Assistant frontend, companion and backend T.Chinese translations.

**Comment by synesthesiam** [comment]
OK, let's split `zh` into multiple dialects. I see in Home Assistant that `zh-Hans` and `zh-Hant` exist for simplified and traditional Chinese. Should we use that, or go with `zh-CN`, `zh-HK`, and `zh-TW`?

**Reply by eddychanwork** [reply]
I think we should go with zh-CN, zh-TW and zh-HK. Because as I mentioned CH, TW and HK have cultural differences, zh-hans and zh-hant could only separate simplified Chinese user and traditional Chinese user.

**Reply by bluefoxlee** [reply]
I kind of second @eddychanwork 's comment. As I understand zh-HK include some extend characters which Taiwan did not use and I am not quite sure how those characters (text) involved in text input etc.  

BTW @eddychanwork. Wonder most HA users in Hong Kong use S.Chinese or T.Chinese in UI.

**Reply by synesthesiam** [reply]
I've renamed the original `zh` to `zh-cn` and added both `zh-hk` and `zh-tw` to the repo :+1: 
Can someone please correct the [native language names](https://github.com/home-assistant/intents/blob/main/languages.yaml) (if necessary)?

**Reply by eddychanwork** [reply]
I believe majority Hong Kong citizens is using either Traditional Chinese and English.
@synesthesiam the native name of zh-HK is 繁體中文（香港）.

**Reply by bluefoxlee** [reply]
PR created for update native names. Thank you @synesthesiam 

@eddychanwork Got it and thanks.

**Reply by synesthesiam** [reply]
Thanks @bluefoxlee! Adding you as `zh-tw` language leader. Would you like to be the leader of the other dialects too? Or @eddychanwork would you be interested in leading `zh-hk`?

**Reply by eddychanwork** [reply]
Thank you for inviting me to be zh-HK language leader. Yes, I would like to pick up this role.

**Reply by bluefoxlee** [reply]
@synesthesiam Thanks again for the help! I can't really speak for zh-hk since I don't speak or understand cantonese. so I will focus on zh-tw for now. :)

@ChienTingLu I would be grateful if you could join zh-tw. ^^

**Reply by bench2012** [reply]
Hi all,

Good to see HA support for local Chinese diag. such as Cantonese. I have been using HA at home for quite a few years now and will be very much like to contribute back to the project. I am from Hong Kong, although I speak Cantonese, English is also my 2nd language, since I worked and live in the UK for about 15 years before I came back to Hong Kong. I also worked in the IT field as a career in Hong Kong (retired now).   

Please let me know if there are any thing I can contribute. Google and all the other "big name" company's translation support for Cantonese sucks BTW ;-)

Bench2012

**Comment by synesthesiam** [comment]
@eddychanwork and @ChienTingLu Please submit two small PR's for your languages, and I will add you as leaders :+1:

**Reply by cmoo2011** [reply]
Ok. Working on it.
Thanks a lot.

**Comment by synesthesiam** [comment]
@swonge @eddychanwork Please coordinate here on `zh-hk`

**Reply by swonge** [reply]
ok

**Comment by synesthesiam** [comment]
Merged https://github.com/home-assistant/intents/pull/1009 from @al-one. Would you be interested in being a language leader for `zh-cn`?

**Reply by al-one** [reply]
Yes, i am interested.

**Reply by synesthesiam** [reply]
Great, thanks! I'm adding you as a leader for `zh-cn` now :slightly_smiling_face:
